Platform/Plugins/com.tle.platform.common/src/com/tle/common/util/ExecUtils.java

```diff
@@ -52,6 +55,71 @@
     "", ".exe", ".bat"
   }; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
 
+  /**
+   * Creates a process which will then kill a given process and it's child processes on Linux.
+   * arg[0] should be the PID of the main process you want to terminate.
+   */
+  public static class LinuxKill {
+    public static void main(String[] args) {
+      try {
+        if (Integer.parseInt(args[0]) > 0) {
+          // get child process PIDs as string
+          Process getChildPid = Runtime.getRuntime().exec("pgrep -P " + args[0]);
+          getChildPid.waitFor();
+          StringBuilder childPid = new StringBuilder();
+          if (getChildPid.exitValue() == 0) {
+            InputStream procIn = getChildPid.getInputStream();
+            int c = 0;
+            while ((c = procIn.read()) != -1) {
+              childPid.append((char) c);
+            }
+          }
+
+          // Kill child process(es)
+          Process childKillProc = Runtime.getRuntime().exec("kill -9 " + childPid);
+          childKillProc.waitFor();
+          if (childKillProc.exitValue() == 0) {
+            LOGGER.debug("Child processes terminated: " + childPid);
+          }
+
+          // kill process itself
+          Process proc = Runtime.getRuntime().exec("kill -9 " + args[0]);
+          int exitVal = proc.waitFor();
+          LOGGER.debug("Exit value: " + exitVal); // often 1 under Ubuntu 12.10
+
+          proc.destroy();
+          childKillProc.destroy();
+          getChildPid.destroy();
+        }
+      } catch (Exception e) {
+        LOGGER.warn("Process Kill failed. Process may be left hanging");
```

Comment:
Be good to know why it failed. (Also, slight change in capitalisation.)

```suggestion
        LOGGER.warn("Process kill failed. Process may be left hanging. " + e.getMessage());
```